onnxruntime/core/optimizer/qdq_transformer/qdq_selector_action_transformer.cc

```diff
@@ -0,0 +1,162 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/optimizer/qdq_transformer/qdq_actions.h"
+#if !defined(ORT_MINIMAL_BUILD)
+#include "core/optimizer/qdq_transformer/qdq_selector_action_transformer.h"
+#include "core/optimizer/qdq_transformer/qdq_selectors.h"
+#endif
+
+namespace onnxruntime {
+
+namespace {
+
+using NTO = onnxruntime::NodesToOptimize;
+
+// create rules for ops that don't change the data
+void DropQDQNodesRules(SelectorsAndActions& qdq_selectors_and_actions) {
+  // 3 nodes. DQ, target, Q. Merge into target and remove DQ and Q.
+  const std::string action_name{"drop"};
+  NTO::NodeLocation dq{NTO::NodeType::kInput, 0};
+  NTO::NodeLocation q{NTO::NodeType::kOutput, 0};
+
+  // Move DQ input 0 to target input 0.
+  // Move Q output 0 to target output 0.
+  std::vector<NodeAndMoveInfo> moves{
+      MoveToSlot(dq, ArgType::kInput, 0, ArgType::kInput, 0),
+      MoveToSlot(q, ArgType::kOutput, 0, ArgType::kOutput, 0)};
+
+  std::unique_ptr<Action> action(new MergeIntoTarget(std::move(moves)));
+
+#if !defined(ORT_MINIMAL_BUILD)
+  std::unique_ptr<NodeSelector> selector(new QDQ::DropDQDNodesSelector());
+  qdq_selectors_and_actions.RegisterSelectorAndAction(action_name,
+                                                      SelectorAndAction::OpVersionsMap{{"Gather", {}},
+                                                                                       {"Reshape", {}},
+                                                                                       {"Transpose", {}},
+                                                                                       {"MaxPool", {12}}},
```

Comment:
It would be better to put those in a central place, like the registry.cc, which clearly shows what ops are supported.

Reply:
This one file defines all the supported ops for the CPU QDQ optimizations. 

It's slightly different to the old setup where it was a 1:1 between an op and a transformer instance. Now it's 1:M between the selector and the ops it is used for. 

I could add a map<name,SelectorAndAction::OpVersionsMap> where the 'name' is the selector name, but that just seems like extra code to put all the op names on consecutive lines. 

Is that significantly better than skimming through this one file to see the supported ops? You lose the adjacency of the ops list to the registration call if we do that though (i.e. I need to go to the registry to find the list of ops that a RegisterSelectorAndAction call is using).

The NNAPI and CoreML QDQ setups will have their own SelectorActionTransformers so the ops they support will be defined there. They will most likely re-use selector instances, but will have different Action implementations.